frontend/src/models/index.js

```diff
@@ -119,17 +118,24 @@ export const RootModel = types
           self.events.put(result);
           self.selectEvent(result.id);
         } catch (error) {
-          console.error(error);
           self.ui.eventList.fetching = false;
           self.ui.eventList.fetchError = 'Could not create event ' + event.name;
         }
       } else {
         // existing event. PUT
         try {
           self.ui.eventList.fetching = true;
-          yield putEvent(event, self.user.token);
-
+          const result = yield putEvent(event, self.user.token);
           self.ui.eventList.fetching = false;
+          event.ticketCatalog.forEach(function(ticketType, index) {
```

Comment:
This for each is just used for iterating the ticketCatalog array. You can just replace it with a normal for loop or even use map instead of forEach here, since you are doing data transformation.